tap_square/client.py

```diff
@@ -52,18 +64,33 @@ def _get_access_token(self):
         return result.body['access_token']
 
     @staticmethod
-    def _get_v2_objects(request_timer_suffix, request_method, body, body_key):
+    @backoff.on_exception(
+        backoff.expo,
+        RetryableError,
+        max_time=180, # seconds
+        on_backoff=log_backoff,
+        jitter=backoff.full_jitter,
+    )
+    def _retryable_v2_method(request_method, body, **kwargs):
+        result = request_method(body, **kwargs)
+
+        if result.is_error():
+            error_message = result.errors if result.errors else result.body
+            if 'Service Unavailable' in error_message:
```

Comment:
For v2 this is the text in the reason that we included in the JIRA for when we should retry, we don't want to retry on any error but we can add more cases here if we want in the future